runners/samza/src/main/java/org/apache/beam/runners/samza/metrics/SamzaMetricsContainer.java

```diff
@@ -99,6 +242,7 @@ public void updateMetrics(String stepName) {
 
     final DistributionUpdater updateDistribution = new DistributionUpdater();
     results.getDistributions().forEach(updateDistribution);
+
     if (commitAllMetricUpdates) {
       stepNameList.stream()
           .map(metricsContainers::getContainer)
```

Comment:
yup we could test this out, possibly turning this optimization on and commitAllUpdates one off. But there is still value for committing all updates periodically since it probably relinquishes some memory every time its called.